src/taco-utils/npmHelper.ts

```diff
@@ -7,76 +7,69 @@
 
 "use strict";
 
-import npm = require("npm");
+import child_process = require("child_process");
+import os = require("os");
 import path = require("path");
 import Q = require("q");
-import nopt = require("nopt");
-import url = require("url");
-import semver = require("semver");
-
-var npmconf = require("npm/lib/config/core.js");
-var configDefs = npmconf.defs;
-var types = configDefs.types;
-var shorthands = configDefs.shorthands;
 
 import installLogLevel = require("./installLogLevel");
 import InstallLogLevel = installLogLevel.InstallLogLevel;
 
 module TacoUtility {
     export class NpmHelper {
-        private static runNpmCommand(npmCommand: string, args: string[], workingDirectory: string, commandFlags?: string[], logLevel?: InstallLogLevel, optionalArgs?: any): Q.Promise<any> {
-            var conf: any = {};
-            if (commandFlags) {
-                conf = nopt(types, shorthands, commandFlags, 0);
-            }
+        private static runNpmCommand(npmCommand: string, args: string[], workingDirectory: string, commandFlags?: string[], logLevel?: InstallLogLevel, silent: boolean = false): Q.Promise<string[]> {
 
-            if (logLevel && logLevel > 0) {
-                conf["logLevel"] = InstallLogLevel[logLevel];
-            }
+            const spawnArgs = [npmCommand].concat(args).concat(commandFlags || []);
 
-            var propertiesToSet: string[] = Object.keys(conf);
-            if (propertiesToSet.indexOf("argv") !== -1) {
-                propertiesToSet.splice(propertiesToSet.indexOf("argv"), 1);
-            }
+            const options = {
+                cwd: workingDirectory,
+                stdio: silent ? "pipe" : "inherit"
+            };
 
-            var oldValues: any = {};
-            return Q.nfcall(npm.load, {}).then(function() {
-                var configList = npm.config.list[0];
+            const npmExecutableName = "npm" + (os.platform() === "win32" ? ".cmd" : "");
 
-                propertiesToSet.forEach(prop => {
-                    oldValues[prop] = configList[prop];
-                    configList[prop] = conf[prop];
-                });
+            const npmProcess = child_process.spawn(npmExecutableName, spawnArgs, options);
```

Comment:
Why are we going back to using npm command? Is there a specific stability issue with using the npm module?